### PR TITLE
Add @osdk/agents package with AgentConfig

### PR DESCRIPTION
**Pull request by mhgbrg**
Library for publishing utils for agents. Very similar to `@osdk/functions`. Currently only exports `AgentConfig` (the equivalent to `FunctionConfig` but for agents), but we expect to add more utils over time.

Generated with Claude Code. I asked it to copy the `@osdk/functions` setup. I reviewed the files and everything looks reasonable, but please let me know if anything looks off.

### CODE REVIEW

File: packages/agents/package.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    "fix-lint": "oxlint -c ./oxlint.config.ts --fix . && oxfmt -c ../../oxfmt.config.ts .",
+    "lint": "oxlint -c ./oxlint.config.ts . && oxfmt -c ../../oxfmt.config.ts --check .",
+    "transpileBrowser": "monorepo.tool.transpile -f esm -m normal -t browser",
+    "transpileCjs": "monorepo.tool.transpile -f cjs -m bundle -t node",
+    "transpileEsm": "monorepo.tool.transpile -f esm -m normal -t node",
+    "transpileTypes": "monorepo.tool.transpile -f esm -m types -t node",
+    "typecheck": "tsc --noEmit --emitDeclarationOnly false"
+  },
+  "peerDependencies": {
+    "@osdk/functions": "workspace:^"
```
Comment (ziyunp) [inline packages/agents/package.json]: Do you want to have dependency on @osdk/function? This means that whenever a user wants to install this package, they also need to install functions package

Comment (mhgbrg) [reply packages/agents/package.json]: The `AgentConfig` type extends the `FunctionConfig` type. I can make it a true dependency instead of a peer dependency if that makes more sense?

Comment (ziyunp) [reply packages/agents/package.json]: Is FunctionConfig a real dependency? Do you want your AgentConfig shape to change whenever FunctionConfig changes?

If not, maybe you can re-define the FunctionConfig type within this repo, so it is not coupled to @osdk/functions.

Comment (mhgbrg) [reply packages/agents/package.json]: Yes, at least for now. We generate functions for agents and need to forward `FunctionConfig` to the generated function.

**Review by ziyunp — DISMISSED** [review]

File: packages/agents/src/scope/types.ts
Comment (mhgbrg) [inline packages/agents/src/scope/types.ts:1]: Copied from functions instead of pulling in with a dep so that we have flexibility to diverge from functions in the future if we want to.

**Review by ziyunp — DISMISSED** [review]

File: .monorepolint.config.mjs [hunk 2]
```diff
@@ -1413,7 +1388,9 @@ function standardPackageRules(shared, options) {
     }),
     fileContents({
       ...shared,
-      excludePackages: ["@osdk/maker"],
+      // @osdk/agents ships a single type and has no tests, so it has no test
```
Comment (nfdiop) [inline .monorepolint.config.mjs]: curious why not let it have the vitest config. Ideally we want all packages to follow the same structure and want to avoid edge cases

File: packages/agents/oxlint.config.ts [hunk 3]
```diff
@@ -14,21 +14,19 @@
...
+// `extends` only carries `rules`/`plugins`/`overrides`, so the root's
+// `ignorePatterns` are re-applied explicitly (otherwise generated/ignored files
+// would start being linted).
+export default defineConfig({
+  extends: [root],
+  ignorePatterns: root.ignorePatterns,
 
-  return resolvedAliases.streams[alias];
-}
+  rules: {},
```
Comment (nfdiop) [inline packages/agents/oxlint.config.ts:31]: Noticed this is different from other packages: we will likely need to be consistent. No actions to take

File: .monorepolint.config.mjs (hunk 2 above)
Comment (mhgbrg) [reply .monorepolint.config.mjs]: Figured it was unnecessary since there were no tests. I don't have strong opinions here. If I need to re-add the config to get this PR merged I'm fine with that.

Comment (mhgbrg) [reply .monorepolint.config.mjs]: Done.

**Review by ssanjay1 — APPROVED** [review]

**Comment by mhgbrg** [comment]
Closing in favor of https://github.com/palantir/osdk-ts/pull/3811.
